build.gradle

```diff
@@ -89,7 +89,7 @@ jenkinsPlugin {
 configurations { forceGroovy }
 
 dependencies {
-    forceGroovy 'org.codehaus.groovy:groovy-all:2.4.8'
+    forceGroovy "org.codehaus.groovy:groovy-all:${groovyVersion}"
```

Comment:
@nickgrealy Can you provide some information why `forceGroovy` was introduced in the first place?

I had a look at when it was introduced in commit 3a973d5c621006ca3b43191fff81a2f932a93d59.
- The first Groovy version used was 2.4.4 - see [here](https://github.com/jenkinsci/groovy-events-listener-plugin/commit/3a973d5c621006ca3b43191fff81a2f932a93d59#diff-503f218d646c10f484fdc9d6315bf2e3R2).
- The first Jenkins version used was 1.581 - see [here](https://github.com/jenkinsci/groovy-events-listener-plugin/commit/3a973d5c621006ca3b43191fff81a2f932a93d59#diff-c197962302397baf3a4cc36463dce5eaR39). It only provided Groovy 1.8.9 - see [here](https://github.com/jenkinsci/jenkins/blob/jenkins-1.581/core/pom.xml#L47).

Therefore I have these questions:
1) Was your intention to use a newer version of Groovy in the plugin than Jenkins provided?
2) Do you think packaging Groovy with the plugin can be removed now that Jenkins provides Groovy 2.4.x itself?

Reply:
Sorry, was on vacation. But from what I remember when digging around in the commit history like you did; it seems that the groovy library that is/was bundled with Jenkins did not include the Grape stuff. So if you used `@Grab` in your script, you would get class not found errors (#21).

Thus I may be wrong, but I think `forceGroovy` thing came from trying to force bundle Groovy with this plugin instead of relying on the one that ships with Jenkins, in order to support Grape. But then that brought along the problem with this issue where if the groovy bundled with this plugin doesn't match the groovy bundled with Jenkins, it fails to load.

Reply:
@nickgrealy is there a chance to also get feedback from you?

Reply:
@krische I suggest to remove everything regarding `forceGroovy` and only use the Groovy version provided by Jenkins which is sufficient. The thing to keep is the Ivy dependency because `@Grab` uses Ivy under the cover. If you would remove it, the `@Grab` test would fail too.

Reply:
Wouldn't `maskClasses = 'groovy.grape'` from `build.gradle` be the thing that prevents using `@Grab`?

Reply:
Yes, this also would have to be changed.

BUT: I quickly checked with removing the lines from `build.gradle` as suggested and running `./gradlew server` (with gradle-jpi-plugin 0.35.0 for another reason) and using the [code](https://github.com/jenkinsci/groovy-events-listener-plugin/blob/master/src/test/groovy/org/jenkinsci/plugins/globalEventsPlugin/integration/IntegrationTest.groovy#L44-L46) from `IntegrationTest`. I doesn't work as intended.

There are some details about classloading in Jenkins in general which have to be understood first. 
- https://groups.google.com/forum/?fromgroups#!topic/job-dsl-plugin/EG6eqQYYI7M
- https://issues.jenkins-ci.org/browse/JENKINS-26377?focusedCommentId=219129&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-219129

Reply:
[JENKINS-26192](https://issues.jenkins-ci.org/browse/JENKINS-26192) is about how `@Grab` support was added to global libraries. It's just a hack. See pull request [#9](https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/9). Therefore I suggest for the moment to stick with the current setup. This means bundling Ivy and in addition also Groovy (for class loader reasons).

Reply:
Another option would be to get rid of the Groovy and Ivy lib completely (including the `maskClasses` and `pluginFirstClassLoader` settings).
Instead add `org.jenkins-ci.plugins.workflow:workflow-cps-global-lib:2.3` as dependency to get `@Grab` support via [GrapeHack.java](https://github.com/jenkinsci/workflow-cps-global-lib-plugin/blob/workflow-cps-global-lib-2.3/src/main/java/org/jenkinsci/plugins/workflow/cps/global/GrapeHack.java) from [workflow-cps-global-lib-plugin](https://github.com/jenkinsci/workflow-cps-global-lib-plugin).
But let's keep this option for later.